### PR TITLE
Fetch transactions from Firebase on load

### PR DESCRIPTION
**Pull request by jasonpetsod**
Remove dummy test transactions passed in as App.props since they're no longer
needed.

Add a static method to create a new Transaction object from transaction data
in Firebase.

Hack tests for App.editTransaction so that they work without fetching data
from Firebase.

Issue: #5

### CODE REVIEW

File: src/Transaction.jsx [hunk 1]
```diff
@@ -7,6 +7,30 @@ const TEST_CATEGORY_ID = 'test-category-id';
 const TEST_PAYEE_ID = 'test-payee-id';
 
 export default class Transaction {
+  static fromFirebaseData(key, data) {
+    let inflow = 0.0;
+    let outflow = 0.0;
+    if (data.amountMinor < 0) {
+      outflow = Math.abs(data.amountMinor) / 100;
```
Comment (witriew) [inline src/Transaction.jsx]: If this conversion happens here, why is `amountMinor` still usable from `Transaction`?

Comment (jasonpetsod) [reply src/Transaction.jsx]: Yeah, this is on a separate branch from 5119bbd1b62f4beaf3df5045c9174eb1a8255116. I propose to submit that other change first and then merge it into this branch?

File: src/App.jsx [hunk 2]
```diff
@@ -15,29 +15,49 @@ export default class App extends React.Component {
...
-      transactions: props.transactions,
+      transactions: {},
     };
 
     this.addTransaction = this.addTransaction.bind(this);
     this.editTransaction = this.editTransaction.bind(this);
   }
 
+  componentDidMount() {
+    return App.budgetRef().child('transactions').once('value')
```
Comment (witriew) [inline src/App.jsx]: Eventually, we should limit this to all transactions within the last 30 days or something like that.

File: src/App.jsx [hunk 3]
```diff
@@ -15,29 +15,49 @@ export default class App extends React.Component {
...
     this.state = {
-      transactions: props.transactions,
+      transactions: {},
     };
 
     this.addTransaction = this.addTransaction.bind(this);
     this.editTransaction = this.editTransaction.bind(this);
   }
 
+  componentDidMount() {
```
Comment (witriew) [inline src/App.jsx:32]: What calls this?

Comment (jasonpetsod) [reply src/App.jsx:32]: React magic: https://facebook.github.io/react/docs/react-component.html#componentdidmount

File: src/App.jsx (hunk 2 above)
Comment (jasonpetsod) [reply src/App.jsx]: Added TODO.

**Review by jasonpetsod — COMMENTED** [review]
PTAL

**Review by witriew — APPROVED** [review]

**Pull request merged**